### PR TITLE
Fix Maven build

### PR DESCRIPTION
**Pull request by kittylyst**
The Maven build was slightly odd & failing OOTB on Java 21 - I think it wasn't picking up the compiler plugin properly.

Also, IntelliJ has Junit 4 on the build path by default, so some old spurious JUnit 4 imports were OK in an IDE build, but failed in a Maven build.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -68,9 +78,12 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.0</version>
```
Comment (JuanMorenoDeveloper) [inline pom.xml]: Maybe we can bump this to the latest `3.13.0`?

File: pom.xml [hunk 2]
```diff
@@ -13,6 +13,13 @@
     <version>0.0.1-SNAPSHOT</version>
     <name>java-concurrent-animated-reboot</name>
     <description>java-concurrent-animated-reboot</description>
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
+        <!-- Library versions -->
+        <junit.version>5.9.2</junit.version>
```
Comment (JuanMorenoDeveloper) [inline pom.xml]: Let's use the latest one `5.11.3`.

File: pom.xml (hunk 1 above)
Comment (kittylyst) [reply pom.xml]: Sure.

File: pom.xml (hunk 2 above)
Comment (kittylyst) [reply pom.xml]: Done.

**Review by JuanMorenoDeveloper — APPROVED** [review]

**Pull request merged**